classes/class-query.php

```diff
@@ -246,12 +246,29 @@ public function query( $args ) {
 		 */
 		$query = apply_filters( 'wp_stream_db_query', $query, $args );
 
-		$result = array();
+		// Build result count query.
+		$count_query = "SELECT COUNT(*) as found
+		FROM $wpdb->stream
+		{$join}
+		WHERE 1=1 {$where}";
+
+		/**
+		 * Filter allows the result count query to be modified before execution.
+		 *
+		 * @param string $query
+		 * @param array  $args
+		 *
+		 * @return string
+		 */
+		$count_query = apply_filters( 'wp_stream_db_count_query', $count_query, $args );
```

Comment:
I don't think we need a filter here.

Reply:
I believe it will be needed, If someone is modifying the select query, they would need to modify the count query as well to make the count match the results shown.